src/helpers/dom.js

```diff
@@ -1,336 +1,292 @@
-        func()
-    } else {
-        insertInfoBox('{{msg.crowdDisabled}}')
-    }
+/*example: www.fastforward.team/hmm?s=53, domain=fastforward.team path=hmm?s=53
+Include subdomain if it's not www. */
+export function crowdQuery(domain, path) {
+  var data = {
+    domain: domain,
+    path: path,
+  };
+  //ff + first 10 characters of SHA256 of fastforward to prevent collisions
+  document.dispatchEvent(
+    new CustomEvent('ff53054c0e13_crowdQuery', { detail: data })
+  );
 }
-
-export function crowdContribute(target, func) {
-    if (typeof func != 'function') {
-        func = () => {
-        }
-    }
-    if (crowdBypassCheckbox.checked && isGoodLink(target)) {
-        docSetAttribute('{{channel.crowd_contribute}}', target)
-        setTimeout(func, 10)
-    } else {
-        func()
-    }
+//target must be a fully qualified URL with protocol
+export function crowdContribute(domain, path, target) {
+  var data = {
+    domain: domain,
+    path: path,
+    target: target,
+  };
+  document.dispatchEvent(
+    new CustomEvent('ff53054c0e13_crowdContribute', { detail: data })
+  );
 }
 
-export function contributeAndNavigate(target) {
-    if (!navigated && isGoodLink(target)) {
-        crowdContribute(target, () => unsafelyNavigate(target))
-    }
-}
+export function unsafelyNavigate(target, referer = null, crowd = false) {
+  //The background script will intercept the request and redirect to html/before-navigate.html or to the target depending on the user's settings.
+  if (navigated) {
+    return;
+  }
+  let url;
+  if (crowd) {
+    url = `https://fastforward.team/bypassed?crowd=true&target=${encodeURIComponent(
+      target
+    )}`;
+  } else {
+    url = `https://fastforward.team/bypassed?target=${encodeURIComponent(
+      target
+    )}`;
+  }
 
-export function unsafelyNavigate(target, referer = null) {
-    //The background script will intercept the request and redirect to html/before-navigate.html or to the target depending on the user's settings.
-    if (navigated) {
-        return
-    }
-    let url = `https://fastforward.team/bypassed?target=${encodeURIComponent(
-        target
-    )}`
-    if (referer) url += `&referer=${encodeURIComponent(referer)}`
+  if (referer) url += `&referer=${encodeURIComponent(referer)}`;
 
-    switch (
-        target //All values here have been tested using "Take me to destinations after 0 seconds."
-        ) {
-        case (/(krnl\.place|hugegames\.io)/.exec(target) || {}).input:
-            url += '&safe_in=15'
-            break
-        case (/(bowfile\.com)/.exec(target) || {}).input:
-            url += '&safe_in=20'
-            break
-        case (/(flux\.li)/.exec(target) || {}).input:
-            url += '&safe_in=25'
-            break
-    }
-    unsafelyAssign(url)
+  switch (
+    target //All values here have been tested using "Take me to destinations after 0 seconds."
+  ) {
+    case (/(krnl\.place|hugegames\.io)/.exec(target) || {}).input:
```

Comment:
We definitely should use the infoBox method to let people know that we don't bypass this website because it's video game cheating,

Reply:
That can be another PR, this one focuses on getting the extension working with MV3